File: application/src/main/java/whenyourcar/application/converter/chat/ChatCommonConverter.java
package whenyourcar.application.converter.chat;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageImpl;
import org.springframework.stereotype.Component;
import whenyourcar.application.converter.car.CarSaleConverter;
import whenyourcar.application.dto.chat.common.ChatCommonResponse;
import whenyourcar.domain.entity.CarSale;
import whenyourcar.domain.entity.Room;
import whenyourcar.domain.entity.User;
import whenyourcar.domain.query.chat.SearchRoomInfoQuery;
import whenyourcar.domain.query.chat.SearchRoomsQuery;

import java.util.List;
import java.util.stream.Collectors;

@Component
@RequiredArgsConstructor
public class ChatCommonConverter {
    private final CarSaleConverter carSaleConverter;
    public ChatCommonResponse.ChatRoomInfoResponseDto toChatRoomResponse(SearchRoomInfoQuery searchRoomInfoQuery) {
        return ChatCommonResponse.ChatRoomInfoResponseDto.builder()
                .roomId(searchRoomInfoQuery.getRoomId())
                .user2Id(searchRoomInfoQuery.getUser2Id())
                .picture(searchRoomInfoQuery.getPicture())
                .name(searchRoomInfoQuery.getName())
                .carSaleInfoDto(carSaleConverter.toCarSaleInfoDto(searchRoomInfoQuery.getCarSale()))
                .build();
    }
}

File: application/src/main/kotlin/howmuchcar/application/converter/chat/ChatCommonConverter.kt
package howmuchcar.application.converter.chat

import howmuchcar.application.converter.car.CarSaleConverter
import howmuchcar.application.dto.chat.ChatRoomCarInfoResponse
import howmuchcar.application.dto.chat.ChatRoomInfoResponse
import howmuchcar.domain.query.chat.SearchRoomInfoQuery
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Component

@Component
@RequiredArgsConstructor
class ChatCommonConverter(
    private val carSaleConverter: CarSaleConverter
) {
    fun toChatRoomResponse(query: SearchRoomInfoQuery) : ChatRoomCarInfoResponse {
        return  ChatRoomCarInfoResponse(
            roomId = query.roomId,
            user2Id = query.user2Id,
            picture = query.picture,
            name = query.name,
            carSaleInfo = carSaleConverter.toCarSaleInfoDto(query.carSale)
        )
    }
}